subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/plugins/JavaGradlePluginPlugin.java

```diff
@@ -182,103 +188,90 @@ private void configureTestKit(Project project, GradlePluginDevelopmentExtension
+                    throw new IllegalArgumentException(String.format(DECLARATION_MISSING_IMPLEMENTATION_MESSAGE, declaration.getName()));
                 }
             }
         });
     }
 
-    private void configureTaskPropertiesValidation(final Project project) {
-        final SourceSet mainSourceSet = project.getConvention().getPlugin(JavaPluginConvention.class).getSourceSets().getByName(SourceSet.MAIN_SOURCE_SET_NAME);
-        final TaskProvider<ValidateTaskProperties> validator = project.getTasks().register(VALIDATE_TASK_PROPERTIES_TASK_NAME, ValidateTaskProperties.class, new Action<ValidateTaskProperties>() {
-            @Override
-            public void execute(ValidateTaskProperties validator) {
-                validator.setGroup(PLUGIN_DEVELOPMENT_GROUP);
-                validator.setDescription(VALIDATE_TASK_PROPERTIES_TASK_DESCRIPTION);
+    private void configurePluginValidations(Project project) {
+        SourceSet mainSourceSet = project.getConvention().getPlugin(JavaPluginConvention.class).getSourceSets().getByName(SourceSet.MAIN_SOURCE_SET_NAME);
+        TaskProvider<Task> naggerTask = project.getTasks().register("nagAboutValidateTaskProperties", task -> {
+            //noinspection Convert2Lambda - Shouldn't use lambdas for task actions
+            task.doFirst(new Action<Task>() {
+                @Override
+                public void execute(Task taskWithAction) {
+                    nagAboutDeprecatedValidateTaskPropertiesTask();
+                }
+            });
+        });
+        TaskProvider<ValidatePlugins> validatorTask = project.getTasks().register(VALIDATE_PLUGINS_TASK_NAME, ValidatePlugins.class, task -> {
+            task.setGroup(PLUGIN_DEVELOPMENT_GROUP);
+            task.setDescription(VALIDATE_PLUGIN_TASK_DESCRIPTION);
 
-                validator.getOutputFile().set(project.getLayout().getBuildDirectory().file("reports/task-properties/report.txt"));
+            task.getOutputFile().set(project.getLayout().getBuildDirectory().file("reports/plugin-development/validation-report.txt"));
 
-                validator.getClasses().setFrom(mainSourceSet.getOutput().getClassesDirs());
-                validator.getClasspath().setFrom(mainSourceSet.getCompileClasspath());
-                validator.dependsOn(mainSourceSet.getOutput());
-            }
+            task.getClasses().setFrom(mainSourceSet.getOutput().getClassesDirs());
+            task.getClasspath().setFrom(mainSourceSet.getCompileClasspath());
+            task.dependsOn(mainSourceSet.getOutput());
+            task.mustRunAfter(naggerTask);
         });
+        project.getTasks().register(VALIDATE_TASK_PROPERTIES_TASK_NAME, org.gradle.plugin.devel.tasks.ValidateTaskProperties.class, validatorTask, (Runnable) JavaGradlePluginPlugin::nagAboutDeprecatedValidateTaskPropertiesTask)
+            .configure(task -> {
+                task.setGroup(PLUGIN_DEVELOPMENT_GROUP);
```

Comment:
I think this task shouldn't have a group, so it doesn't show up with `tasks`. I think the description should be re-added to the nagger task, since the description alone doesn't make the task show up in `tasks`. Only the group controls this.

Reply:
Fair point.